packages/apps/docs/src/docs/reference/kadena-client/client-utils.md

```diff
@@ -0,0 +1,273 @@
+---
+title: Kadena client utilities
+description:
+  The `@kadena/client-utils` library provides a TypeScript based application programming interface API for interacting with smart contracts and the Kadena network. The library includes helper functions for the `coin` module that you can import using `@kadena/client-utils/coin` and `core` functions that you can use to develop interfaces for custom contracts.
+menu: Reference
+label: Kadena client
+order: 4
+layout: full
+tags: ['TypeScript', 'Kadena client', 'frontend']
+---
+
+# Kadena client utilities
+
+The `@kadena/client-utils` library provides a TypeScript based application programming interface API for interacting with smart contracts and the Kadena network. 
+The library includes helper functions for the `coin` module that you can import using `@kadena/client-utils/coin`.
+For example, you can import specific `coin` module functions into a TypeScript program with the following statement:
+
+```typescript
+import { getBalance, transferCrossChain } from "@kadena/client-utils/coin"
+```
+  
+The library also exports `core` utilities that you can use to develop your own API functions for any type kind of smart contract.
+
+The client utilities are listed alphabetically in this reference to make they easy to find by name.
+However, you typically call the utilities in the following order to represent the normal workflow:
```

Comment:
Actually its not correct. there is no order and each function works independently.
`asyncPipe` is a general functional programming helper that let you to pipe (compose) some functions together. 
I already used it to create other helpers (`submitClient`, `preflightClient`, `dirtyReadClient`, `dirtyReadClient`)
the use case of asyncPipe is creating new utilities for example I am working on `queryAllChainsClient` that let you execute a command on all chains in parallel.
Regarding xClient (e.g. submitClient), each function usually manages the whole workflow from singing to submitting and listening for the result of a command. so a very high level API is like this
`config->command->result`